### DISCUSSION TITLE
Using your instructions I accidentally typed "talosctl gen config" instead of "talosctl machineconfig patch"

### DISCUSSION THREAD

**Discussion by jlowe64** [siderolabs/talos#6845] [Q&A] [answered]
I was in the wrong directory and was trying to install cilium and I used the instruction `talosctl gen config` to generate a config with cni set to none to prep for a cilium install, however I realised after the fact that this in fact overwrote my already generated config, that had all my certificates adn connection data in the `talosconfig` file. I was looking through the commands for `talos get mc`, but it doesn't seem to have the ability to get the config back to where it was before. 

I know this is a lesson learned for me and I might have to destroy my cluster, but I was hoping there was some way to recover from this without doing that. 

The error I get after the regeneration accident is:

`rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority (possibly because of \"x509: Ed25519 verification failure\" while trying to verify candidate authority certificate \"talos\")"`

I assume this is because the command I issued overwrote the controlplane.yaml,  talosconfig , and worker.yaml with something new that doesn't have the original keypairs in mind?

Any help would be appreciated!

**Comment by smira** [comment, marked as answer]
If you have `kubeconfig` somewhere, you could try running a privileged pod in the host mount namespace and recovering machine config from `STATE` partition (`/system/state`). 

Once you have machine config for the controlplane node, you can follow https://www.talos.dev/v1.3/talos-guides/configuration/managing-pki/

**Comment by steverfrancis** [comment]
We've also made https://github.com/siderolabs/talos/issues/6847

**Reply by jlowe64** [reply]
That's awesome! I only have a few hours of learning Talos, so I was just sorta clicking around anyway before I codified things. But would be nice to have that guard rail in place!

**Reply by DmitriyMV** [reply]
It's now implemented on main branch, and will likely be in 1.4

**Reply by jlowe64** [reply]
@DmitriyMV that's amazing. Thank you for your quick contribution to your tool on my behalf!
